src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,94 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int MULTIPLIER = 2;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > threshold) {
+            resize();
+        }
+        int index = getIndex(key);
+        Node<K, V> current = table[index];
+        while (current != null) {
+            if (Objects.equals(current.key, key)) {
+                current.value = value;
+                return;
+            }
+            if (current.next == null) {
+                current.next = new Node<>(key, value, null);
+                size++;
+                return;
+            }
+            current = current.next;
+        }
+        table[index] = new Node<>(key, value, null);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> result;
+        return (result = getNode(key)) == null ? null : result.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        int newLength = table.length * MULTIPLIER;
+        Node<K, V>[] oldArray = table;
+        table = new Node[newLength];
+        size = 0;
+        Node<K, V> current;
+        for (Node<K, V> kvNode : oldArray) {
+            current = kvNode;
+            while (current != null) {
+                put(current.key, current.value);
```

Comment:
Using `put` inside the `resize` method can cause unnecessary resizing and incorrect size increment. Consider directly transferring nodes to the new table without using `put` to avoid these issues.